waspc/src/Wasp/Lib.hs

```diff
@@ -37,64 +39,74 @@ compile ::
   CompileOptions ->
   IO ([CompileWarning], [CompileError])
 compile waspDir outDir options = do
-  (analyzerWarnings, appSpecOrAnalyzerErrors) <- analyzeWaspProject waspDir options
+  appSpecOrAnalyzerErrors <- analyzeWaspProject waspDir options
+  analyzerWarnings <- warnIfDotEnvPresent waspDir
   compilerWarningsAndErrors <- case appSpecOrAnalyzerErrors of
     Left analyzerErrors -> return ([], toList analyzerErrors)
     Right appSpec ->
       case ASV.validateAppSpec appSpec of
         [] -> do
           (generatorWarnings, generatorErrors) <- Generator.writeWebAppCode appSpec outDir (sendMessage options)
           return (map show $ generatorWarningsFilter options generatorWarnings, map show generatorErrors)
-        validationErrors -> do
-          return ([], map show validationErrors)
+        validationErrors -> return ([], map show validationErrors)
   return $ (analyzerWarnings, []) <> compilerWarningsAndErrors
 
 analyzeWaspProject ::
   Path' Abs (Dir WaspProjectDir) ->
   CompileOptions ->
-  IO ([CompileWarning], Either (NonEmpty CompileError) AS.AppSpec)
```

Comment:
No worries, it's still in :)

I've moved it to the `Lib.hs#compile` function because `analyzeWaspProject` was only used in two places - `Lib.hs#compile` and `Deps.hs#deps` (which was ignoring it).

Or did you perhaps notice this and mean "we should have the warnings in this function (and not in `compile` for semantic purposes?"

Reply:
Oof, sorry! 🤦🏻‍♂️ Totally missed where it was moved too. Yeah, I think that is fine there. Resolving this 👍🏻